core/src/main/scala/io/snappydata/SnappyThinConnectorTableStatsProvider.scala

```diff
@@ -0,0 +1,92 @@
+/*
+ * Changes for SnappyData data platform.
+ *
+ * Portions Copyright (c) 2017 SnappyData, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+
+package io.snappydata
+
+import java.sql.{CallableStatement, Connection}
+import java.util.{Properties, Timer, TimerTask}
+
+import scala.collection.JavaConversions
+
+import com.gemstone.gemfire.CancelException
+import com.pivotal.gemfirexd.internal.engine.ui.{SnappyIndexStats, SnappyRegionStats}
+import io.snappydata.Constant._
+
+import org.apache.spark.SparkContext
+import org.apache.spark.sql.SmartConnectorHelper
+import org.apache.spark.sql.execution.datasources.jdbc.JdbcUtils
+
+object SnappyThinConnectorTableStatsProvider extends TableStatsProviderService {
+
+  private var conn: Connection = null
+  private var getStatsStmt: CallableStatement = null
+  private var _url: String = null
+
+  def initializeConnection(): Unit = {
+    conn = JdbcUtils.createConnectionFactory(
+      _url + ";route-query=false;" , new Properties())()
+    getStatsStmt = conn.prepareCall("call sys.GET_SNAPPY_TABLE_STATS(?)")
+  }
+
+  def start(sc: SparkContext, url: String): Unit = {
+    _url = url
+    initializeConnection()
+    val delay = sc.getConf.getLong(Constant.SPARK_SNAPPY_PREFIX +
+        "calcTableSizeInterval", DEFAULT_CALC_TABLE_SIZE_SERVICE_INTERVAL)
+    doRun = true
+    new Timer("SnappyThinConnectorTableStatsProvider", true).schedule(
+      new TimerTask {
+        override def run(): Unit = {
+          try {
+            if (doRun) {
+              aggregateStats()
+            }
+          } catch {
+            case _: CancelException => // ignore
+            case e: Exception => logError("SnappyThinConnectorTableStatsProvider", e)
+          }
+        }
+      }, delay, delay)
+  }
+
+  def executeStatsStmt(): Unit = {
+    if (conn == null) initializeConnection()
+    getStatsStmt.registerOutParameter(1, java.sql.Types.BLOB)
+    getStatsStmt.execute()
+  }
+
+  override protected def getStatsFromAllServers: (Seq[SnappyRegionStats], Seq[SnappyIndexStats]) = {
+    try {
+      executeStatsStmt()
+    } catch {
+      case e: Exception =>
+        logWarning("SnappyThinConnectorTableStatsProvider: exception while retrieving stats " +
+            "from Snappy embedded cluster. Check whether the embedded cluster is stopped ")
+        logDebug("Exception stack trace: ", e)
```

Comment:
Perhaps logWarn? That can log just the e.toString() while full trace can be with logDebug.